accounting/formatters/ChtcScheddCpuOspoolFormatter.py

```diff
@@ -0,0 +1,171 @@
+from .BaseFormatter import BaseFormatter
+from datetime import datetime
+from collections import OrderedDict
+
+
+def hhmm(hours):
+    # Convert float hours to HH:MM
+    h = int(hours)
+    m = int(60 * (float(hours) - int(hours)))
+    return f"{h:02d}:{m:02d}"
+
+
+def handle_dashes(dtype, fmt, value):
+    # Cast value to dtype and format it.
+    # Right-align any non-castable values and return them as is.
+    formatted_str = "<td></td>"
+    try:
+        value = dtype(value)
+        formatted_str = f"<td>{value:{fmt}}</td>"
+    except ValueError:
+        formatted_str = f"""<td style="text-align: right">{value}</td>"""
+    except Exception as err:
+        print(f"Caught unexpected exception {str(err)} when converting {value} to {repr(dtype)}.", file=sys.stderr)
+        formatted_str = f"""<td style="text-align: right">{value}</td>"""
+    return formatted_str
+
+
+class ChtcScheddCpuOspoolFormatter(BaseFormatter):
+
+    def get_table_title(self, table_file, report_period, start_ts, end_ts):
+        info = self.parse_table_filename(table_file)
+        # Format date(s)
+        start = datetime.fromtimestamp(start_ts)
+        if report_period in ["daily"]:
+            start_date = start.strftime("%Y-%m-%d")
+            title_str = f"CHTC per {info['agg'].rstrip('s')} usage for jobs completed on <strong>{start_date}</strong> on the OSPool"
+        elif report_period in ["weekly", "monthly"]:
+            end = datetime.fromtimestamp(end_ts)
+            start_date = start.strftime("%Y-%m-%d")
+            end_date = end.strftime("%Y-%m-%d")
+            title_str = f"CHTC per {info['agg'].rstrip('s')} usage for jobs completed from <strong>{start_date} to {end_date}</strong> on the OSPool"
+        else:
+            end = datetime.fromtimestamp(end_ts)
+            start_date = start.strftime("%Y-%m-%d %H:%M:%S")
+            end_date = end.strftime("%Y-%m-%d %H:%M:%S")
+            title_str = f"CHTC per {info['agg'].rstrip('s')} usage for jobs completed from <strong>{start_date} to {end_date}</strong> on the OSPool"
+        return title_str
+
+    def get_subject(self, report_period, start_ts, end_ts, **kwargs):
+        # Format date(s)
+        start = datetime.fromtimestamp(start_ts)
+        if report_period in ["daily", "weekly", "monthly"]:
+            start_date = start.strftime("%Y-%m-%d")
+            subject_str = f"CHTC AP on OSPool {report_period.capitalize()} Usage Report {start_date}"
```

Comment:
We should ask for some feedback from the RCFs on the email subject, but this is good for now.